distributed/tests/test_worker_state_machine.py

```diff
@@ -1002,6 +1005,195 @@ def test_gather_priority(ws):
     ]
 
 
+def test_constrained_task_handles_resources_on_success(ws_with_running_task):
+    ws = ws_with_running_task
+    assert ws.available_resources == {"R": 0}
+    # Ensure that x transitions through the ``constrained`` state
+    assert_story(
+        ws.story("x"),
+        [
+            ("x", "compute-task", "released"),
+            ("x", "released", "waiting", "waiting", {"x": "ready"}),
+            ("x", "waiting", "ready", "waiting", {"x": "constrained"}),
+            ("x", "waiting", "constrained", "constrained", {"x": "executing"}),
+            ("x", "constrained", "executing", "executing", {}),
+        ],
+        strict=False,
+    )
+
+    ws.handle_stimulus(
+        ExecuteSuccessEvent(
+            key="x",
+            value=None,
+            start=0.0,
+            stop=1.0,
+            nbytes=8,
+            type=None,
+            stimulus_id="success",
+        ),
+    )
+    assert ws.tasks["x"].state == "memory"
+    assert ws.available_resources == {"R": 1}
+
+
+@pytest.mark.xfail(reason="distributed#6565")
```

Comment:
`XFAIL` to be tackled in a follow-up PR. #6565 is already tracking this issue.

Reply:
This is already green

Reply:
Good catch, somehow missed that.